source/DistanceAndDirection/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ViewModels/ProCircleViewModel.cs

```diff
@@ -158,37 +187,138 @@ private double TravelRateInSeconds
+                }
+            }
+        }
+
         double travelTime = 0.0;
         /// <summary>
         /// Property for time display
         /// </summary>
         public double TravelTime
         {
+            
             get
             {
                 return travelTime;
             }
             set
             {
                 if (value < 0.0)
+                {
+                    UpdateFeedbackWithGeoCircle();
+                    ClearTempGraphics();
+                    if (HasPoint1)
+                        // Re-add the point as it was cleared by ClearTempGraphics() but we still want to see it
+                        AddGraphicToMap(Point1, ColorFactory.GreenRGB, null, true, 5.0);
                     throw new ArgumentException(DistanceAndDirectionLibrary.Properties.Resources.AEMustBePositive);
+                }
 
                 travelTime = value;
+                
+                double distanceInMeters = TravelRateInSeconds * TravelTimeInSeconds;
+                if (RateUnit != DistanceTypes.Meters)
+                {
+                    // Prevent graphical glitches from excessively high inputs
+                    distanceInMeters = ConvertFromTo(RateUnit, DistanceTypes.Meters, TravelRateInSeconds * TravelTimeInSeconds);
+                }
+                if (distanceInMeters > DistanceLimit)
+                {
+                    RaisePropertyChanged(() => TravelTimeString);
+                    UpdateDistance(TravelRateInSeconds * TravelTimeInSeconds, RateUnit, false);
+                    ClearTempGraphics();
+                    if (HasPoint1)
+                        // Re-add the point as it was cleared by ClearTempGraphics() but we still want to see it
+                        AddGraphicToMap(Point1, ColorFactory.GreenRGB, null, true, 5.0);
+                    throw new ArgumentException(DistanceAndDirectionLibrary.Properties.Resources.AEInvalidInput);
+                }
 
                 // we need to make sure we are in the same units as the Distance property before setting
-                UpdateDistance(TravelRateInSeconds * TravelTimeInSeconds, RateUnit);
+                UpdateDistance(TravelRateInSeconds * TravelTimeInSeconds, RateUnit, true);
 
-                RaisePropertyChanged(() => TravelTime);
+                // Trigger validation to clear error messages as necessary
+                RaisePropertyChanged(() => RateTimeUnit);
+                RaisePropertyChanged(() => TimeUnit);
+                RaisePropertyChanged(() => TravelRateString);
+                RaisePropertyChanged(() => TravelTimeString);
             }
+
         }
 
-        private void UpdateDistance(double distance, DistanceTypes fromDistanceType)
+        private void UpdateDistance(double distance, DistanceTypes fromDistanceType, bool belowLimit)
         {
-            if(CircleType == CircleFromTypes.Diameter)
```

Comment:
Believe this fixes: https://github.com/Esri/distance-direction-addin-dotnet/issues/324